airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -511,18 +512,23 @@ public KubePodProcess(final boolean isOrchestrator,
     // This is safe only because we are blocking the init pod until we copy files onto it.
     // See the ExitCodeWatcher comments for more info.
     exitCodeFuture = new CompletableFuture<>();
-    podWatch = fabricClient.resource(podDefinition).watch(new ExitCodeWatcher(
+    sharedInformerFactory = fabricClient.informers();
+    final SharedIndexInformer<Pod> podInformer = sharedInformerFactory.sharedIndexInformerFor(Pod.class, 30 * 1000L);
```

Comment:
A couple of things here. Should the `sharedIndexInformerFor` call be a `sharedIndexInformerForCustomResource ` instead so we can use an `OperationContext` with both the name and namespace of the resource? Then we won't need to be reading all pod info (an expensive operation) on a long cadence. 

Also, if we're re-introducing a timeout for this we'll need to inject it with env vars so we can use different values for the user-interactive pod processes.

Reply:
these all seem like good ideas. I'm having trouble getting this to work at all (acceptance tests are failing because the informer seems to never actually get triggered?) but will incorporate them once I make this work at all

the 30 seconds is the resync period, i.e. [how frequently the informer rebuilds its cache](https://rohaan.medium.com/introduction-to-fabric8-kubernetes-java-client-informer-api-b945082d69af). So it's more about how long we'll take to detect pod state updates (I think).

Reply:
Yeah, that's why it needs to be configurable, for fast jobs that service UI requests that we don't want to wait for 1min extra.

Reply:
got it. What's the typical best practice for adding an env var? (i.e. do I need to  add it to EnvConfigs + pass the EnvConfig down into the KubePodProcess or something)